### DISCUSSION TITLE
Default Encryption Key for ADE Could Not be Found

### DISCUSSION THREAD

**Discussion by ourichermath** [apprenticeharper/DeDRM_tools#1591] [Q&A]
In a nutshell, I can't add my Adobe Digital Editions 2.0.1 encryption key to the DeDRM 7.1.0 plugin in Calibre 5.12.0 under Linux Mint 20.1 Cinnamon.

Calibre is running natively under Linux.  ADE is in a Crossover (Wine-ish) bottle.  I've successfully authorized ADE with my credentials.  In Calibre > Preferences > Plugins > File Type > DeDRM > Customize Plugin > Adobe Digital Editions ebooks, I've added the WINEPREFIX of "~/.cxoffice" (actually the full, expanded path instead of the shortcut) which is where the installed Crossover bottles are).  When I click the green "+" to add the default ADE key, I get:

"DeDRM 7.1.0:  Getting Default Adobe Digital Editions Key
The default encryption key for Adobe Digital Editions could not be found."

Searching the internet, I've found a few vague references to running the adobekey.py file (which on my system is under ~/.config/calibre/plugins/DeDRM/libraryfiles).  But, I have no idea how.  I've also seen references to issues involving Calibre/DeDRM running natively while ADE is in a bottle/wine.  But, again, I have no idea how to go about fixing this.

Can anyone give me, or point me to, detailed, current instructions for how to resolve this?

**Comment by ElleKayEm** [comment]
Probably the easiest way is to temporarily install Windows calibre with DeDRM where you have ADE installed.  Add a book from ADE to calibre.  Then you can save the key from the Windows calibre/DeDRM setup, and import it into your Linux calibre/DeDRM settings.  Then you can uninstall the Windows calibre.

**Reply by ivandiv** [reply]
This would be the solution, but how do you install Calibre for Wine? I tried downloading the Windows versions -- normal, 64 bit and portable. Only the portable worked when going "wine calibre-portable....exe"  because its an .exe file. It installed, but failed to run as it couldn't import some files, it said.

**Reply by ElleKayEm** [reply]
Hmmm, maybe Windows calibre 4.23 and DeDRM 6.8.1 would work under wine.  I don't know.  The current script is here:

https://github.com/apprenticeharper/DeDRM_tools/blob/master/DeDRM_plugin/adobekey.py

But again that method would require the appropriate version of Windows Python and I'm not sure what else.

**Comment by ivandiv** [comment]
Furthermore, I managed to download the ineptkey.pyw script by I <3 Cabbages from here:
https://www.ismoothblog.com/2012/12/how-to-remove-drm-from-kobo-vox-ereader.html

Reading through the file, it says:
    "This script only supports Windows and Mac OS X.  For Linux "
            "you should be able to run ADE and this script under Wine (with "
            "an appropriate version of Windows Python installed).")
as this is from 2012 (sic!) I'm not even going to try that.

Any other suggestions?
Thank you!
